pandas/core/generic.py

```diff
@@ -1872,12 +1872,17 @@ def _drop_labels_or_levels(self, keys, axis: int = 0):
     # ----------------------------------------------------------------------
     # Iteration
 
-    def __hash__(self) -> int:
+    def _hash(self) -> int:
         raise TypeError(
             f"{repr(type(self).__name__)} objects are mutable, "
             f"thus they cannot be hashed"
         )
 
+    # https://github.com/python/typeshed/issues/2148#issuecomment-520783318
+    # Incompatible types in assignment (expression has type "None", base class
+    # "object" defined the type as "Callable[[object], int]")
+    __hash__: None = _hash  # type: ignore[assignment]
```

Comment:
this seems sketchy, but i dont have any better ideas

Reply:
i agree this seems like a crazy work-around. we shouldn't be paving the way with mypy incompatiblities.

Reply:
If it's OK to drop the current custom error message, then would
```
__hash__: None. # type: ignore[assignment]
```
(as suggested https://github.com/python/typeshed/issues/2148#issuecomment-520783318) be OK? If so, I'll do that later